plugins/quetz_mamba_solve/quetz_mamba_solve/api.py

```diff
@@ -0,0 +1,70 @@
+import json
+
+from fastapi import APIRouter
+from fastapi.responses import PlainTextResponse
+
+from .solver import MambaSolver
+
+router = APIRouter()
+
+
+def parse_list(names):
+    """
+    accepts strings formatted as lists with square brackets
+    names can be in the format
+    "[bob,jeff,greg]" or '["bob","jeff","greg"]'
+    """
+
+    def remove_prefix(text: str, prefix: str):
+        if text.startswith(prefix):
+            text = text[len(prefix) :]  # noqa: E203
+        return text
+
+    def remove_postfix(text: str, postfix: str):
+        if text.endswith(postfix):
+            text = text[: -len(postfix)]
+        return text
+
+    if names is None:
+        return
+
+    # we already have a list, we can return
+    if isinstance(names, list):
+        return names
+
+    # if we don't start with a "[" and end with "]" it's just a normal entry
+    if not names.startswith("[") and not names.endswith("]"):
+        return [names]
+
+    names = remove_prefix(names, "[")
+    names = remove_postfix(names, "]")
+
+    names_list = names.split(",")
+    names_list = [remove_prefix(n.strip(), "\"") for n in names_list]
+    names_list = [remove_postfix(n.strip(), "\"") for n in names_list]
+
+    return names_list
+
+
+@router.get(
+    "/api/mamba/solve/{channels}/{subdir}/{spec}", response_class=PlainTextResponse
+)
+def mamba_solve(channels, subdir, spec):
+    channels = parse_list(channels)
+    spec = parse_list(spec)
+    s = MambaSolver(channels, subdir)
```

Comment:
it would be cool to have a long-living solver where repodata is only loaded again when it's necessary. There is a replace_channel functionality that should work for this case.

Reply:
This is done through the time expiry based LRU cache.